oap-ape/ape-java/ape-client/src/main/java/com/intel/oap/ape/service/netty/client/ResponseHandler.java

```diff
@@ -41,26 +44,42 @@
     private final Object boolResponseLock = new Object();
     private NettyMessage.BooleanResponse recentBoolResponse;
 
+    private boolean hasNextBatch = true;
+    private int sequenceId = 0;
+    private final PriorityQueue<NettyMessage.ReadBatchResponse> receivedBatches;
+    private final LinkedBlockingQueue<NettyMessage.ReadBatchResponse> availableBatches;
+
+    private boolean closed = false;
+
+    public ResponseHandler() {
+        receivedBatches = new PriorityQueue<>(
+                Comparator.comparingInt(NettyMessage.ReadBatchResponse::getSequenceId));
+        availableBatches = new LinkedBlockingQueue<>();
+    }
+
     @Override
     protected void channelRead0(ChannelHandlerContext ctx, NettyMessage msg) throws Exception {
         try {
             Class<?> msgClazz = msg.getClass();

```

Comment:
just curious, is this best practice?

Reply:
I think this is not bad. We can find the same implementation in Flink when it's doing type checking.